src/parser/roc_video_parser.cpp

```diff
@@ -77,8 +77,9 @@ rocDecStatus RocVideoParser::MarkFrameForReuse(int pic_idx) {
     if (pic_idx < 0) {
         return ROCDEC_INVALID_PARAMETER;
     }
-    //todo::
-    return ROCDEC_NOT_IMPLEMENTED;
+
+    decode_buffer_pool_[pic_idx].use_status &= ~kFrameUsedForDisplay;
```

Comment:
Hi @jeffqjiangNew, this won't work in case of multi-threading. We can only flush out the buffer after it has been consumed. Otherwise the display buffer will get corrupted in async operation. The idea is the decoder can't use this buffer again until it is fully consumed. Does shifting the output_pic_list, assume this?

Reply:
output_pic_list_ can only be managed by the parser. It stores the buffers that get dumped from DPB. It shouldn't be touched by the consumer. Lets discuss in the meeting to clarify our thoughts.